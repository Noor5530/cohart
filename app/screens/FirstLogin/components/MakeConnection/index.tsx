import React from "react";
import {
  View,
  Text,
  useWindowDimensions,
  TouchableOpacity,
  Image,
} from "react-native";

import ArrowRight from "assets/Arrow_Right.png";
import styles from "./style";

interface IProps {
  onPressButton(): void
}

const Discover: React.FC<IProps> = ({onPressButton}: IProps) => {
  const { width, height } = useWindowDimensions();
  return (
    <View style={[styles.container, { width, height }]}>
      <Text style={styles.headingText}>Ready to make connections?</Text>
      <Text style={styles.descriptionText}>
        Tell us about yourself to start meeting members of the community
      </Text>
      <TouchableOpacity
        style={[styles.buttonStyle, styles.primaryButton]}
        onPress={onPressButton}
      >
        <Text style={[styles.buttonText]}>Let&apos;s go</Text>
        <Image source={ArrowRight} />
      </TouchableOpacity>
    </View>
  );
};

export default Discover;
