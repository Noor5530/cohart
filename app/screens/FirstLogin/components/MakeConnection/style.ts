import { appFonts } from "components/text";
import { primary } from "config/colors";
import { StyleSheet } from "react-native";
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from "react-native-responsive-screen";

const styles = StyleSheet.create({
  container: {
    backgroundColor: "white",
  },
  headingText: {
    fontFamily: appFonts.InterBold,
    fontSize: hp("6"),
    marginLeft: wp("10"),
    marginBottom: hp("7"),
    marginTop: hp("15"),
  },
  descriptionText: {
    fontFamily: appFonts.InterRegular,
    fontSize: hp("2.8"),
    width: wp("70"),
    marginLeft: wp("10"),
    marginBottom: hp("14"),
  },
  buttonStyle: {
    height: 36,
    width: wp("40"),
    borderRadius: 24.01,
    justifyContent: "center",
    alignItems: "center",
    flexDirection: "row",
    alignSelf: "center",
  },
  primaryButton: {
    backgroundColor: primary,
  },
  buttonText: {
    textTransform: "uppercase",
    fontFamily: appFonts.InterBold,
    color: "black",
    marginRight: wp("3"),
  },
});

export default styles;
