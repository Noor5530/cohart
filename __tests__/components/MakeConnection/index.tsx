import "react-native";
import React from "react";
import renderer from "react-test-renderer";

import MakeConnection from "../../../app/screens/FirstLogin/components/MakeConnection";
jest.useFakeTimers();

test("renders correctly", () => {
  const tree = renderer.create(<MakeConnection onPressButton={jest.fn}/>).toJSON();

  expect(tree).toMatchSnapshot();
});
