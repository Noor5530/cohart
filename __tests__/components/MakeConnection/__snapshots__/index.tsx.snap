// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
      },
      Object {
        "height": 1334,
        "width": 750,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "fontFamily": "Inter-Bold",
        "fontSize": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginTop": undefined,
      }
    }
  >
    Ready to make connections?
  </Text>
  <Text
    style={
      Object {
        "fontFamily": "Inter-Regular",
        "fontSize": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "width": undefined,
      }
    }
  >
    Tell us about yourself to start meeting members of the community
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#E6FF00",
        "borderRadius": 24.01,
        "flexDirection": "row",
        "height": 36,
        "justifyContent": "center",
        "opacity": 1,
        "width": undefined,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "black",
            "fontFamily": "Inter-Bold",
            "marginRight": undefined,
            "textTransform": "uppercase",
          },
        ]
      }
    >
      Let's go
    </Text>
    <Image
      source={
        Object {
          "testUri": "../../../app/assets/Arrow_Right.png",
        }
      }
    />
  </View>
</View>
`;
